### PR TITLE
Add citation optional parameter to abstract_citations and retrieval

### PR DESCRIPTION
**Pull request by MrShepardd**
There is not enough functionality for downloading with the exception of self-citation. 
It would be nice to add an extra parameter that is supported by the Citation Overview API.

### CODE REVIEW

**Comment by Amadest** [comment]
LGTM!

File: pybliometrics/scopus/abstract_citations.py [hunk 1]
```diff
@@ -148,6 +148,10 @@ def __init__(self, eid, start, end=datetime.now().year, refresh=False):
             The last year for which the citation count should be loaded.
             Default is the current year.
 
+        citation : str (optional, default='')
+            Allows for the exclusion of self-citations.
+            The default behavior is to include all citations.
+
```
Comment (Michael-E-Rose) [inline pybliometrics/scopus/abstract_citations.py:154]: Please add accepted values

File: pybliometrics/scopus/abstract_citations.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def volume(self):
         """Volume for the abstract."""
         return self._citeInfoMatrix.get('volume')
 
-    def __init__(self, eid, start, end=datetime.now().year, refresh=False):
+    def __init__(self, eid, start, end=datetime.now().year, citation='', refresh=False):
```
Comment (Michael-E-Rose) [inline pybliometrics/scopus/abstract_citations.py:136]: Can we make the default `None`?

File: pybliometrics/scopus/abstract_citations.py [hunk 3]
```diff
@@ -191,7 +197,7 @@ def __str__(self):
         if len(authors) > 1:
             authors[-1] = " and ".join([authors[-2], authors[-1]])
         s = f"Document '{self.title}' by {', '.join(authors)}\npublished in "\
-            f"'{self.publicationName}' has the following citation trajectory "\
+            f"'{self.publicationName}' has the following {self._citation} citation trajectory "\
```
Comment (Michael-E-Rose) [inline pybliometrics/scopus/abstract_citations.py:200]: How would this look? "has the following exclude-self citation trajectory"? Doesn't look readable. What about a dictionary approach, mapping the two accepted values to labels used in the __str__? One could combine this with a check whether the provided values are among those the Scopus API accepts

**Review by Michael-E-Rose — CHANGES_REQUESTED** [review]
Very good idea, thanks for the initiative!

I request some few changes. Do you also want to add some information in the corresponding example page? https://github.com/pybliometrics-dev/pybliometrics/blob/master/docs/examples/CitationOverview.rst No hard feelings if not, I gladly update it

File: pybliometrics/scopus/abstract_citations.py (hunk 2 above)
Comment (MrShepardd) [reply pybliometrics/scopus/abstract_citations.py]: Of course, changed in recent commits

File: pybliometrics/scopus/abstract_citations.py (hunk 1 above)
Comment (MrShepardd) [reply pybliometrics/scopus/abstract_citations.py]: Added

File: pybliometrics/scopus/abstract_citations.py (hunk 3 above)
Comment (MrShepardd) [reply pybliometrics/scopus/abstract_citations.py]: I think "publication has the following citation trajectory without self-citations" or "publication has the following citation trajectory without books" should look more readable, changed that in the last commit.

**Comment by MrShepardd** [comment]
I also added parameter information on the example page.

**Comment by Amadest** [comment]
@Michael-E-Rose Hello! Is this information enough? We are waiting for you to make these updates to change our upload algorithms, relying on your library =)
Scopus is able to issue data on the citation rate of the document for each document by years of the type `year: cited_count`, including excluding self-citation. This is useful for analyzing specific documents and authors. It is one thing when an article has 25 citations and 24 of them are self-citations, and quite another when an article has 25 citations, of which 0 are self-citations. This already says a lot.

**Review by Michael-E-Rose — APPROVED** [review]

**Comment by Michael-E-Rose** [comment]
Yes your e right, I try to be faster. Especially when the PR comes out of the community, which I value very very highly!

I'll write tests late on. There should be a new minor release soon.

**Pull request merged**